ui/components/component-library/banner/banner.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { BannerBase, Icon, ICON_NAMES } from '..';
+
+import {
+  COLORS,
+  SEVERITIES,
+  SIZES,
+} from '../../../helpers/constants/design-system';
+import { BANNER_SEVERITIES } from './banner.constants';
+
+export const Banner = ({
+  children,
+  className,
+  severity = SEVERITIES.INFO,
+  ...props
+}) => {
+  const severityIcon = () => {
+    switch (severity) {
+      case SEVERITIES.DANGER:
+        return (
+          <Icon
+            name={ICON_NAMES.DANGER_FILLED}
+            size={SIZES.LG}
+            color={COLORS.ERROR_DEFAULT}
+          />
+        );
+      case SEVERITIES.WARNING:
+        return (
+          <Icon
+            name={ICON_NAMES.WARNING_FILLED}
+            size={SIZES.LG}
+            color={COLORS.WARNING_DEFAULT}
+          />
+        );
+      case SEVERITIES.SUCCESS:
+        return (
+          <Icon
+            name={ICON_NAMES.CHECK_CIRCLE_ON_FILLED}
+            size={SIZES.LG}
+            color={COLORS.SUCCESS_DEFAULT}
+          />
+        );
+      // Defaults to SEVERITIES.INFO
+      default:
+        return (
+          <Icon
+            name={ICON_NAMES.INFO_FILLED}
+            size={SIZES.LG}
+            color={COLORS.PRIMARY_DEFAULT}
+          />
+        );
+    }
+  };
```

Comment:
Could you save some lines of code here by returning an object and spreading it to the Icon component? Something like:

```jsx
  const severityIcon = () => {
    switch (severity) {
      case SEVERITIES.DANGER:
        return {
          name: ICON_NAMES.DANGER_FILLED,
          color: COLORS.ERROR_DEFAULT,
        };
      case SEVERITIES.WARNING:
        return {
          name: ICON_NAMES.WARNING_FILLED,
          color: COLORS.WARNING_DEFAULT,
        };
      case SEVERITIES.SUCCESS:
        return {
          name: ICON_NAMES.CHECK_CIRCLE_ON_FILLED,
          color: COLORS.SUCCESS_DEFAULT,
        };
      // Defaults to SEVERITIES.INFO
      default:
        return {
          name: ICON_NAMES.INFO_FILLED,
          color: COLORS.PRIMARY_DEFAULT,
        };
    }
  };

return (
    <BannerBase
      startAccessory={<Icon size={SIZES.LG} {...severityIcon()} />}
      backgroundColor={severityBackground()}
```

```